server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -97,14 +111,34 @@ public ClientInfoResource(
 
   @GET
   @Produces(MediaType.APPLICATION_JSON)
-  public Iterable<String> getDataSources()
+  public Iterable<String> getDataSources(@Context final HttpServletRequest request)
   {
-    return getSegmentsForDatasources().keySet();
+    if (authConfig.isEnabled()) {
+      // This is an experimental feature, see - https://github.com/druid-io/druid/pull/2424
```

Comment:
instead of repeating this line everywhere, can you just put it in one central place e.g. DruidAuthModule or AuthorizationInfo?